backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py

```
@@ -0,0 +1,225 @@
+        try:
+            zip_file_handle = open(zip_path, 'wb')
+            zip_file_handle.write(file_content)
+        except IOError as e:
+            logger.error(f"Error writing zip file {zip_path}: {e}")
+            raise HTTPException(status_code=500, detail=f"Failed to save uploaded zip: {e}")
+        
+        # Extract zip file using utility function
+        try:
+            extract_zip(zip_path, extract_dir)
+        except zipfile.BadZipFile as e:
+            logger.error(f"Invalid zip file: {e}")
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)